### DISCUSSION TITLE
DEPRECATION WARNING: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

### DISCUSSION THREAD

**Discussion by badaczewski** [gatsbyjs/gatsby#39146] [Ideas / Feature Requests] [answered]
When building production JavaScript and CSS bundles I get multiples warnings about Gatsby using the legacy API. Gatsby needs to give us a way to use the modern API, and/or hide the warnings.

⠏ Building HTML renderer
[=========================== ]   28.587 s 134/136 99% Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs
Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

⠧ Building HTML renderer
[=========================== ]   28.587 s 134/136 99% Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs
Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

⠋ Building HTML renderer
[=========================== ]   28.587 s 134/136 99% Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs
Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

⠸ Building HTML renderer
[=========================== ]   28.587 s 134/136 99% Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs
Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

⠼ Building HTML renderer
[=========================== ]   28.587 s 134/136 99% Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs
Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

⠏ Building HTML renderer
[=========================== ]   28.587 s 134/136 99% Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs
Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

**Comment by rickpoulin2** [comment]
Finally figured this out. Configure as follows in your gatsby-config.js:

```js
module.exports = {
  ...
  plugins: [
    ...
    {
      resolve: "gatsby-plugin-sass",
      options: {
        sassOptions: {
          api: "modern"
        }
      }
    },
    ...
  ]
}
```

**Reply by badaczewski** [reply, marked as answer]
That didn't remove the legacy warnings for me, but adding an addtional option did.

```typescript
    {
      resolve: `gatsby-plugin-sass`,
      options: {
        sassOptions: {
          api: "modern",
          silenceDeprecations: ['legacy-js-api'],
        },
      },
    },
```

**Reply by johnrackles** [reply]
Per the documentation I set the "api" to "modern-compiler" which removed the warnings without setting silenceDeprecations.

**Reply by rt-joe** [reply]
This didn't work for me either. The `api` option is _outside_ of the options passed to the underlying sass-loader via `sassOptions` (see [plugin source](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sass/src/gatsby-node.js) and [sass-loader options](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sass/src/gatsby-node.js)). This plugin would need to accept the `api` option and pass it to the correct property in the loader (i.e. `api` and not through `sassOptions`).
